dataland-internal-storage/src/main/kotlin/db/migration/V14__MigrateSubstantialContributionToClimateChangeAdaptation.kt

```diff
@@ -0,0 +1,62 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.getOrJavaNull
+import db.migration.utils.getOrJsonNull
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.json.JSONObject
+
+/**
+ * This migration script updates the existing SFDR datasets, more specifically it introduces an additional layer to
+ * each sector in the high-impact climate sectors
+ */
+class V14__MigrateSubstantialContributionToClimateChangeAdaptation : BaseJavaMigration() {
+    private val mapOfOldToNewFieldNames = mapOf(
+        "substantialContributionToClimateChangeAdaptionInPercent"
+            to "substantialContributionToClimateChangeAdaptationInPercent",
+    )
+    private val mapOfOldToNewdDnshFieldName = mapOf(
+        "dnshToClimateChangeAdaption" to "dnshToClimateChangeAdaptation",
+    )
+
+    /**
+     * Migrates substantialContributionToClimateChangeAdaption field to substantialContributionToClimateChangeAdaptation
+     */
+    @Suppress("NestedBlockDepth")
+    fun migrateSubstantialContributionToClimateChangeAdaptation(dataTableEntity: DataTableEntity) {
+        val euTaxoDataSet = JSONObject(dataTableEntity.companyAssociatedData.getString("data"))
+        listOf("revenue", "capex", "opex").forEach { cashFlowType ->
+            mapOfOldToNewFieldNames.forEach {
+                val euTaxoDataSubsetCashFlowType = euTaxoDataSet.getOrJavaNull(cashFlowType) ?: return@forEach
+                euTaxoDataSubsetCashFlowType as JSONObject
+                euTaxoDataSubsetCashFlowType.put(
+                    it.value,
+                    euTaxoDataSubsetCashFlowType.getOrJsonNull(it.key),
+                )
+                euTaxoDataSubsetCashFlowType.remove(it.key)
+                if (euTaxoDataSubsetCashFlowType.getOrJavaNull("alignedActivities") != null) {
+                    euTaxoDataSubsetCashFlowType.getJSONArray("alignedActivities").forEach { actitivy ->
+                        actitivy as JSONObject
+                        actitivy.put(it.value, actitivy[it.key])
+                        actitivy.remove(it.key)
+                        actitivy.put(
+                            mapOfOldToNewdDnshFieldName.values.first(),
```

Comment:
Also die geschichte mit dem first hier ist super weird. Die maps haben ja beide nur ein element. "A" to "B" definiert ein Pair in kotlin. Lass die Map doch weg und weise einfach `val x = "A" to "B"` zu. Da sparst du dir auch ein bisschen nestedness mit.

Reply:
@stewal19

Reply:
Gute Frage. Muss ich auch erst nochmal reingucken, da ich das auch nicht geschrieben habe

Reply:
Ah ok sorry. Dachte du hättest das geschrieben.   Vllt an den Autor weiterleiten, weil der schneller versteht, was zu tun ist?

Reply:
@exatar Kannst du dir das vll anschauen?

Reply:
Ok.. ich gucke es mir an

Reply:
die zweite map wurde durch ein pair ersetzt.. jedoch bleiben first und second erhalten, um auf die elemente des paares zuzugreifen

Reply:
In dem Kontext ist das aber fine.